src/enosys.c

```diff
@@ -504,6 +504,18 @@ int fi_no_cntr_wait(struct fid_cntr *cntr, uint64_t threshold, int timeout)
 {
 	return -FI_ENOSYS;
 }
+uint64_t fi_no_cntr_readerr(struct fid_cntr *cntr)
```

Comment:
This no op function should have the same return with its corresponding op, `fi_cntr_readerr` will return `uint64_t`, so we have to use `uint64_t` here as well. Otherwise the function pointer will be a mismatched type.